src/components/ProtectedRoute/ProtectedRoute.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { Navigate } from 'react-router-dom';
+
+const ProtectedRoute = ({ children, requiredAuthLevel }) => {
+    // Placeholder to actually check if user is authenticated
+    const useAuth = () => {
+        if (Math.random() > 0.5) {
+            return true;
+        }
+        return false;
+    }
+
+    const user = useAuth();
+
+    switch (requiredAuthLevel) {
+        case "user":
+            break;
+        case "anonymous":
+            return <Navigate to="/login" />;
```

Comment:
For anonymous can you make it so that if a user session exists they are redirected to the home page? The use case here would be if a logged in user tries to access the login/signup page they should be redirected.